docs/logging.md

```diff
@@ -0,0 +1,160 @@
+---
+id: logging
+title: Logging
+---
+
+### Logging
+
+Almost every application nowadays does logging, whether it's logging of business events, history of some state of 
+application or even debug record sequences. We log to files, distributed systems, standard outputs of OSes, and we 
+got used to it so much it seems really simply to us. Many existing libraries provide many ways of logging from 
+your application's code, either automatic or manual.
+
+### What's wrong?
+
+First, in fact, logging is a side effect. Whether we log anything to a standard output, or a file - it's an interaction 
+with the outside world ('outside' relative to our program). Thus, logging as a process can be represented as an effect and operations - as an algebra.  
+Second, logging itself seems easy, but stuff around it - not so much. MDC, contexts, structured logging - all these 
+things are very helpful on a day-to-day basis, helping us to read and write our logs, making them clean, comprehensible 
```

Comment:
"helpful... helping"
Slightly disturbing repetition, maybe replace with a synonim?
From the top of my head: "helpful" -> "handy", or "helping" -> "assisting", "aiding"